PiBox.Plugins/Persistence/EntityFramework/src/PiBox.Plugins.Persistence.EntityFramework/EntityFrameworkPlugin.cs

```diff
@@ -32,5 +34,17 @@ public void ConfigureApplication(IApplicationBuilder applicationBuilder)
         {
             DiagnosticListener.AllListeners.Subscribe(new DiagnosticObserver());
         }
+
+        public void ConfigureHealthChecks(IHealthChecksBuilder healthChecksBuilder)
```

Comment:
This logic appears to be buggy as it looks for an IDbContext to register the EF health checks but inside the DI-Extensions class the IHealthChecksBuilder Extension method only accepts a TContext of type DbContext, not the interface. 
See:
https://github.com/sia-digital/pibox/blob/main/PiBox.Plugins/Persistence/EntityFramework/src/PiBox.Plugins.Persistence.EntityFramework/DependencyInjectionExtensions.cs#L29

This results in an exception as follows when starting a pibox dependent service that uses the EFPlugin: 
_System.Security.VerificationException: Method PiBox.Plugins.Persistence.EntityFramework.DependencyInjectionExtensions.AddEfContext: type argument 'Some.IExampleDbContext' violates the constraint of type parameter 'TContext'.
   at System.RuntimeMethodHandle.GetStubIfNeeded(RuntimeMethodHandleInternal method, RuntimeType declaringType, RuntimeType[] methodInstantiation)
   at System.Reflection.RuntimeMethodInfo.MakeGenericMethod(Type[] methodInstantiation)_

Consider changing the logic here or adding the interface as a valid type for the Extension method.

Reply:
the problem was, that it found the interface too.
I didnt filter for classes :dagger: